sorting/bubble_sort.rb

```diff
@@ -1,19 +1,21 @@
 def bubble_sort(array)
-  n = array.length
-  loop do
-    swapped = false
-
-    (n - 1).times do |i|
-      if array[i] > array[i + 1]
-        array[i], array[i + 1] = array[i + 1], array[i]
-        swapped = true
+  array_length = array.length
+  return array if array_length <= 1
+  unsorted_until_index = array_length - 1
+  sorted = false
+  until sorted
+    sorted = true
+    for i in (0..unsorted_until_index)
+      if array[i+1]
+        if array[i] > array[i+1]
+          array[i], array[i+1] = array[i+1], array[i]
+          sorted = false
+        end
       end
     end
```

Comment:
Suggestion: I think [an index check](https://github.com/TheAlgorithms/Ruby/pull/186/files#diff-0ca68e1c2f3cb3b62610801c9f5751b1c10225212263365927998e84c71b15e0R9) can be removed with [exclusive range literal](https://ruby-doc.org/core-3.0.3/Range.html#:~:text=those%20created%20using%20...%20exclude%20the%20end%20value.).

```suggestion
    for i in (0...unsorted_until_index)
      if array[i] > array[i+1]
        array[i], array[i+1] = array[i+1], array[i]
        sorted = false
      end
    end
```

---

Another way:

```suggestion
    0.upto(unsorted_until_index - 1) do |i|
      if array[i] > array[i+1]
        array[i], array[i+1] = array[i+1], array[i]
        sorted = false
      end
    end
```

Reply:
I have changed the code according to the suggestion